app/controllers/reviews_controller.rb

```diff
@@ -19,12 +16,12 @@ def create
   end
 
   def edit
-    @review = current_user.reviews.find(params[:id])
-    @buddy = Buddy.find(params[:buddy_id])
+    @review = Review.find(params[:id])
+    @buddy = @review.buddy_id
```

Comment:
This is a bit misleading: `@buddy` doesn't contain a buddy object but an ID. 

Is this instance variable even needed? You can access the buddy using `@review.buddy` in the view.